packages/components/src/components/DistributionChart.tsx

```diff
@@ -23,29 +24,88 @@ export const DistributionChart: React.FC<DistributionChartProps> = ({
   height,
   width,
 }: DistributionChartProps) => {
+  let [isLogX, setLogX] = React.useState(false);
+  let [isExpY, setExpY] = React.useState(false);
   const [sized, _] = useSize((size) => {
     let shape = distribution.pointSet();
-    let widthProp = width !== undefined ? width - 20 : size.width - 10;
     if (shape.tag === "Ok") {
-      return (
-        <div>
-          <SquiggleVegaChart
-            data={{ con: shape.value.continuous, dis: shape.value.discrete }}
-            width={widthProp}
-            height={height}
-            actions={false}
-          />
-        </div>
-      );
+      let spec = buildSpec(isLogX, isExpY, shape.value);
+      if (spec.tag == "Ok") {
+        let widthProp = width ? width - 20 : size.width - 10;
+        var result = (
+          <div>
+            <Vega
+              spec={spec.value}
+              data={{ con: shape.value.continuous, dis: shape.value.discrete }}
+              width={widthProp}
+              height={height}
+              actions={false}
+            />
+          </div>
+        );
+      } else {
+        var result = (
+          <ErrorBox heading={spec.value.heading}>{spec.value.error}</ErrorBox>
+        );
+      }
     } else {
-      return (
-        <div>
-          <ErrorBox heading="Distribution Error">
-            {distributionErrorToString(shape.value)}
-          </ErrorBox>
-        </div>
+      var result = (
+        <ErrorBox heading="Distribution Error">
+          {distributionErrorToString(shape.value)}
+        </ErrorBox>
       );
     }
+    return (
+      <>
+        {result}
+        <div>
+          <CheckBox label="Log X scale" value={isLogX} onChange={setLogX} />
```

Comment:
It would be neat if this were grayed out or similar when not accessible, though it would be tricky to add in the message why as well. Don't need to do this for this PR.

Reply:
I could do a tooltip! That would be pretty easy and small